### PR TITLE
[MM-12746] Migrate user update actions from user_actions.jsx to redux

### PR DESCRIPTION
**Pull request by MirlanMaksv**
#### Summary
Migrated to redux actions according to ticket task, updated/created tests

#### Ticket Link
[MM-12746](https://github.com/mattermost/mattermost-server/issues/10148)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Touches critical sections of the codebase (password changing component)

### CODE REVIEW

File: components/user_settings/display/manage_timezones/manage_timezones.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                automaticTimezone: props.automaticTimezone,
+            }};
+
+        expect(props.actions.updateMe).toHaveBeenCalled();
+        expect(props.actions.updateMe).toHaveBeenCalledWith(expected);
+
+        expect(props.updateSection).toHaveBeenCalled();
+        expect(props.updateSection).toHaveBeenCalledWith('');
+    });
+});
```
Comment (saturninoabril) [inline components/user_settings/display/manage_timezones/manage_timezones.test.js]: 💯

File: components/admin_console/system_users/system_users_dropdown/system_users_dropdown.jsx [hunk 2]
```diff
@@ -85,9 +88,10 @@ export default class SystemUsersDropdown extends React.Component {
         };
     }
 
-    handleMakeActive = (e) => {
+    handleMakeActive = async (e) => {
         e.preventDefault();
-        updateActive(this.props.user.id, true, null, this.props.onError);
+        await this.props.actions.updateUserActive(this.props.user.id, true).
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply components/admin_console/system_users/system_users_dropdown/system_users_dropdown.jsx]: Same with other instances below.

File: components/user_settings/general/user_settings_general.test.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            updateMe: jest.fn(),
+            sendVerificationEmail: jest.fn(),
+            setDefaultProfileImage: jest.fn(),
+            uploadProfileImage: jest.fn(),
+        },
+        maxFileSize: 1024,
+    };
+
+    test('submitUser() should have called updateMe', () => {
+        const updateMe = jest.fn(() => Promise.resolve());
```
Comment (saturninoabril) [inline components/user_settings/general/user_settings_general.test.js]: Resolving with `undefined` value will cause Jest test warning like [UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'data' of undefined](https://build.mattermost.com/blue/organizations/jenkins/mw%2Fmattermost-webapp/detail/PR-2361/1/pipeline/32/#step-33-log-137).  Please return with resolved value.  Empty object may work but preferably either with data or error. You may also use convenience function from Jest - `mockResolvedValue`.
```suggestion
        const updateMe = jest.fn().mockResolvedValue({data: true});
```

File: components/user_settings/general/user_settings_general.test.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        expect(uploadProfileImage).toHaveBeenCalledTimes(1);
+        expect(uploadProfileImage).toHaveBeenCalledWith(requiredProps.user.id, mockFile);
+
+        expect(wrapper.state('pictureFile')).toBe(null);
+        expect(wrapper.instance().submitActive).toBe(false);
+
+        expect(requiredProps.updateSection).toHaveBeenCalledTimes(1);
+        expect(requiredProps.updateSection).toHaveBeenCalledWith('');
+    });
+});
```
Comment (saturninoabril) [inline components/user_settings/general/user_settings_general.test.js]: 💯

File: components/admin_console/system_users/system_users_dropdown/system_users_dropdown.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MirlanMaksv) [reply components/admin_console/system_users/system_users_dropdown/system_users_dropdown.jsx]: I was using `async/await` for implicit promise return, with that I was `await`ing execution completions of functions and callbacks in `.then` block, it was used only in tests. I made requested changes and for some reason tests are still passing. I am surprised )

File: components/user_settings/advanced/user_settings_advanced.jsx [hunk 5]
```diff
@@ -140,7 +141,7 @@ export default class AdvancedSettingsDisplay extends React.Component {
 
         this.setState({isSaving: true});
 
-        await actions.savePreferences(userId, preferences);
+        actions.savePreferences(userId, preferences);
```
Comment (saturninoabril) [inline components/user_settings/advanced/user_settings_advanced.jsx]: I think this async/await for `handleSubmit` is needed so that the new preference will reflect at min view setting.  Could you please revert this unrelated change?

Comment (MirlanMaksv) [reply components/user_settings/advanced/user_settings_advanced.jsx]: oops, my inattention. I have reverted

**Comment by DSchalla** [comment]
Thanks for your PR, @MirlanMaksv!
@saturninoabril Since it was assigned to you, I didn't do the merge myself. :-)

**Comment by saturninoabril** [comment]
@DSchalla Oh I missed to remove it.  Nothing special, it's only my way of not missing notification from this PR :)

@MirlanMaksv Thank you for your contribution!

**Comment by MirlanMaksv** [comment]
It is a pleasure to contribute to such a great app, thank you for your support :)
